netty/src/main/java/io/grpc/transport/netty/BufferingHttp2ConnectionEncoder.java

```diff
@@ -67,6 +69,44 @@
  * in replacement for {@link io.netty.handler.codec.http2.DefaultHttp2ConnectionEncoder}.
  */
 class BufferingHttp2ConnectionEncoder extends DecoratingHttp2ConnectionEncoder {
+
+  /**
+   * Thrown if buffered streams are terminated due to this encoder being closed.
+   */
+  public static final class ChannelClosedException extends Http2Exception {
```

Comment:
Why do these extend Http2Exception? I'm not sure it works well, given that normally we use Http2Exceptions for determining what to _send_ but here we are using them for informing of receipt. That makes the Http2Errors hard to specify, since we don't want to communicate an error to the remote.